site/src/lib/s3.ts

```diff
@@ -0,0 +1,88 @@
+import { S3Client } from "@aws-sdk/client-s3";
+import * as envalid from "envalid";
+
+let s3Client: S3Client | undefined;
+
+export const getS3Client = (): S3Client => {
+  if (!s3Client) {
+    const env = envalid.cleanEnv(process.env, {
+      S3_API_ENDPOINT: envalid.str(),
+      S3_BUCKET: envalid.str(),
+      S3_ACCESS_KEY_ID: envalid.str(),
+      S3_SECRET_ACCESS_KEY: envalid.str(),
+    });
+
+    s3Client = new S3Client({
+      endpoint: env.S3_API_ENDPOINT,
+      region: "us-east-1",
+      credentials: {
+        accessKeyId: env.S3_ACCESS_KEY_ID,
+        secretAccessKey: env.S3_SECRET_ACCESS_KEY,
+      },
+      forcePathStyle: true,
+    });
+  }
+
+  return s3Client;
+};
+
+let s3Bucket: string | undefined;
+
+export const getS3Bucket = (): string => {
+  if (!s3Bucket) {
+    const env = envalid.cleanEnv(process.env, {
+      S3_BUCKET: envalid.str(),
+    });
+
+    s3Bucket = env.S3_BUCKET;
+  }
+
+  return s3Bucket;
+};
+
+let s3PathPrefix: string | undefined;
+
+const getS3PathPrefix = (): string => {
+  if (!s3PathPrefix) {
+    const env = envalid.cleanEnv(process.env, {
+      NEXT_PUBLIC_VERCEL_ENV: envalid.str({ default: "" }),
+      VERCEL_GIT_COMMIT_REF: envalid.str({ default: "" }),
+    });
+
+    s3PathPrefix = `/${
+      env.NEXT_PUBLIC_VERCEL_ENV === "preview" ? env.VERCEL_GIT_COMMIT_REF : ""
+    }`;
+  }
+
+  return s3PathPrefix;
+};
+
+export const resolveS3ResourcePath = (
+  category: "blocks" | "avatars",
+  subpath: string,
+): string => {
+  return `${getS3PathPrefix()}/${category}/${subpath}`;
```

Comment:
I am not sure we need two buckets in any foreseeable future. It will complicate the setup but won't bring any practical benefits. Even if we have two separate Node.js apps, they can still connect to a single bucket for quite a while. I updated Wrangler setup in https://github.com/blockprotocol/blockprotocol/pull/565/commits/7eb49c9dc4e36354dcb408f71bc891e9f3456b83 – good point about the name. WDYT?